bottomless/src/replicator.rs

```diff
@@ -1402,38 +1388,33 @@ impl Replicator {
                     }
                 }
                 let frame = self.get_object(key.into()).send().await?;
-                let mut frameno = first_frame_no;
                 let mut reader = BatchReader::new(
-                    frameno,
+                    first_frame_no,
                     tokio_util::io::StreamReader::new(frame.body),
                     self.page_size,
                     compression_kind,
                 );
 
                 while let Some(frame) = reader.next_frame_header().await? {
-                    let pgno = frame.pgno();
-                    let page_size = self.page_size;
+                    last_received_frame_no = reader.next_frame_no();
                     reader.next_page(&mut page_buf).await?;
                     if self.verify_crc {
                         checksum = frame.verify(checksum, &page_buf)?;
                     }
-                    pending_pages.insert(pgno, &page_buf).await?;
-                    if frame.is_committed() {
-                        let pending_pages = std::mem::replace(
-                            &mut pending_pages,
-                            TransactionPageCache::new(
```

Comment:
If `TransactionPageCache` is not longer used (I think it's the only place), we can remove it as dead code completely.

Reply:
I temporarily left it to keep bottomless-cli compiling. bottomless-cli just needs encryption support too, but I'd rather do it in a follow-up